lib/presentation/controller/recipes_controller.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:get/get.dart';
+
+import '../../core/util/api_constants.dart';
+import '../../data/model/recipe_model.dart';
+import '../../domain/usecase/i_usecase.dart';
+import '../../domain/usecase/implementations/get_random_recipes_usecase.dart';
+
+class RecipesController extends GetxController
+    with StateMixin<List<RecipeModel>> {
+  final UseCase getRandomRecipesUseCase = Get.find<GetRandomRecipesUseCase>();
+
+  RxList<RecipeModel> listOfRecipes = RxList<RecipeModel>();
+
+  @override
+  void onInit() {
+    super.onInit();
+    getRandomRecipes(ApiConstants.numberOfRandomRecipes);
+  }
+
+  void getRandomRecipes(int numberOfRecipes) async {
+    listOfRecipes.clear();
+    change(null, status: RxStatus.loading());
+    try {
+      listOfRecipes.addAll(
+        await getRandomRecipesUseCase.call(
+          params: numberOfRecipes,
+        ),
+      );
+      change(listOfRecipes, status: RxStatus.success());
```

Comment:
Is it possible that at some point we can get to get an empty list?

Reply:
It wouldn't make sense to return an empty list because the service always asks for a fixed amount of recipes, so the API should provide the specified amount, nevertheless I handle that case in the controller now, in case of an unexpected circumstance.